src/ui/public/vislib/lib/dispatch.js

```diff
@@ -39,6 +39,48 @@ export function VislibLibDispatchProvider(Private, config) {
       this._listeners = {};
     }
 
+    pieClickResponse(data) {
+      const points = [];
+
+      let dataPointer = data;
+      while (dataPointer && dataPointer.rawData) {
+        points.push(dataPointer.rawData);
+        dataPointer = dataPointer.parent;
+      }
+
+      if (data.rawData.table.$parent) {
+        const { table, column, row, key } = data.rawData.table.$parent;
+        points.push({ table, column, row, value: key });
+      }
+
+      return points;
+    }
+
+    seriesClickResponse(data) {
+      const points = [];
+
+      ['xRaw', 'yRaw', 'zRaw', 'seriesRaw', 'rawData', 'tableRaw'].forEach(val => {
+        if (data[val]) {
+          points.push(data[val]);
+        }
+      });
+
+      return points;
+    }
+
+    clickEventResponse(d) {
+      const _data = d3.event.target.nearestViewportElement ?
+        d3.event.target.nearestViewportElement.__data__ : d3.event.target.__data__;
+      const isSlices = !!(_data && _data.slices);
+
+      const data = d.input || d;
+
+      return {
+        e: d3.event,
+        data: isSlices ? this.pieClickResponse(data) : this.seriesClickResponse(data),
+      };
+    }
+
```

Comment:
It would be helpful to include some comments here about each of these new methods, just to make their purpose clearer, and so it is understood whether any of them are "public" or just implementation details of this class. (From what I can tell they appear to be the latter).

Reply:
yeah, just internal stuff